File: backend/notifications/src/main/java/com/oxygensend/notifications/domain/Notification.java
package com.oxygensend.notifications.domain;

import java.time.LocalDateTime;
import java.util.UUID;

public record Notification(UUID id,
                           String title,
                           String content,
                           String recipient,
                           Channel channel,
                           LocalDateTime createdAt,
                           LocalDateTime sentAt) {
}

File: backend/notifications/src/main/kotlin/com/oxygensend/notifications/domain/Notification.kt
package com.oxygensend.notifications.domain

import java.time.LocalDateTime
import java.util.UUID

data class Notification(
    val id: UUID,
    val title: String,
    val content: String,
    val recipient: String,
    val channel: Channel,
    val createdAt: LocalDateTime,
    val sentAt: LocalDateTime
)